src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,1931 @@
+
+        string targetType(string inner)
+            => langVersion > LanguageVersion.CSharp12 ? $"System.ReadOnlySpan<{inner}[]>" : $"{inner}[][]";
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Implicit_NullableAnalysis_Outer(LanguageVersion langVersion)
+    {
+        var source = """
+            #nullable enable
+            class C
+            {
+                System.Span<string>? M1(string[] arg) => arg;
+                System.ReadOnlySpan<string>? M2(string[] arg) => arg;
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (4,26): error CS9244: The type 'Span<string>' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter 'T' in the generic type or method 'Nullable<T>'
+            //     System.Span<string>? M1(string[] arg) => arg;
+            Diagnostic(ErrorCode.ERR_NotRefStructConstraintNotSatisfied, "M1").WithArguments("System.Nullable<T>", "T", "System.Span<string>").WithLocation(4, 26),
+            // (5,34): error CS9244: The type 'ReadOnlySpan<string>' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter 'T' in the generic type or method 'Nullable<T>'
+            //     System.ReadOnlySpan<string>? M2(string[] arg) => arg;
+            Diagnostic(ErrorCode.ERR_NotRefStructConstraintNotSatisfied, "M2").WithArguments("System.Nullable<T>", "T", "System.ReadOnlySpan<string>").WithLocation(5, 34));
+    }
+
+    [Fact]
+    public void Conversion_Array_Span_Implicit_NullableAnalysis_ExtensionMethodReceiver()
+    {
+        var source = """
+            #nullable enable
+            static class C
+            {
+                static void MS(string[] a, string?[] b)
+                {
+                    a.M1(); b.M1();
+                    a.M2(); b.M2();
+                    a.M3(); b.M3();
+                    a.M4(); b.M4();
+                }
+                static void M1(this System.Span<string> arg) { }
+                static void M2(this System.Span<string?> arg) { }
+                static void M3(this System.ReadOnlySpan<string> arg) { }
+                static void M4(this System.ReadOnlySpan<string?> arg) { }
+                static void MI(int[] a, int?[] b)
+                {
+                    a.M5(); b.M5();
+                    a.M6(); b.M6();
+                    a.M7(); b.M7();
+                    a.M8(); b.M8();
+                }
+                static void M5(this System.Span<int?> arg) { }
+                static void M6(this System.Span<int> arg) { }
+                static void M7(this System.ReadOnlySpan<int?> arg) { }
+                static void M8(this System.ReadOnlySpan<int> arg) { }
+            }
```

Comment:
>} [](http://example.com/codeflow?start=12&length=1)

Are we testing co- and contra-variance for extension method receiver in initial binding (rather than nullable analysis)?

Perhaps test:
```csharp
static class Extensions
{
    static void M1<T>(this Span<T> arg) { }
    static void M2<T>(this ReadOnlySpan<T> arg) { }
}
class C<T, U>
{
    static void F1(T[] a)
    {
        a.M1<U>();
        a.M2<U>();
    }
}
```
... with:
- `where U : T`
- `where T : class where U : class, T`
- `where T : class, U where U : class`